security/advancedtls/advancedtls_integration_test.go

```diff
@@ -404,17 +404,24 @@ func TestEnd2End(t *testing.T) {
 			}
 			s := grpc.NewServer(grpc.Creds(serverTLSCreds))
 			defer s.Stop()
-			go func(s *grpc.Server) {
+			errChan := make(chan error, 1)
+			go func(errChan chan error, s *grpc.Server) {
```

Comment:
`net.Listen()` isn't blocking. So it doesn't need to happen in the goroutine.
The error returned from `s.Serve()` isn't much helpful. Can just ignore that.

This goroutine can just for `Serve()`

```go
lis, err := net.Listen()
if err != nil { t.Fatal() }
pb.RegisterGreeterServer(s, &serverImpl{})	
go s.Serve(lis)
```

Reply:
Thanks for the suggestion! It looks like I have to open another PR ... Sorry for causing the mess ....